### PR TITLE
Config attribute to allow undeclared checked exceptions to be thrown …

### LINKED ISSUE #28987: Provide configuration attribute to allow checked exceptions from @Transactional interceptor
Add a transaction service config attribute (throwCheckedExceptions) which reverts behavior to pre https://github.com/OpenLiberty/open-liberty/issues/26171

### PR DESCRIPTION
**Pull request by jonhawkes**
…from @Transactional methods

#build

fixes #28987

### CODE REVIEW

**Review by joe-chacko — DISMISSED** [review]
Recommend `s/isThrowCheckedExceptions/canThrowCheckedExceptions/`.

Otherwise, lgtm!

**Comment by jonhawkes** [comment]
#libby
#build

**Comment by wraschke** [comment]
#libby
#build

File: dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -16,6 +16,9 @@
 #NLS_ENCODING=UNICODE
 #NLS_MESSAGEFORMAT_NONE
 #
+throw.checked.exceptions=Throw checked exceptions
+throw.checked.exceptions.desc=Specifies whether undeclared checked exceptions are allowed to be thrown from methods annotated with @Transactional.
```
Comment (helyarp) [inline dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties]: To replace the "are allowed to be" wording, change the desc to "Specifies whether methods annotated with `@Transactional` can throw undeclared checked exceptions."

Note: I had to add backticks to keep the annotation from changing to use a lowercase t.

**Review by helyarp — COMMENTED** [review]
Suggested change

**Comment by jonhawkes** [comment]
fixes #28987

**Comment by jonhawkes** [comment]
#libby
#build

File: dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (jonhawkes) [reply dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties]: Not sure what you mean about the backticks. Is it ok now?

**Review by helyarp — COMMENTED** [review]
Looks good!

File: dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (helyarp) [reply dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties]: @jonhawkes Yes, the change is ok. Thx for making the change promptly. I added the **ID reviewed** label.

**Review by joe-chacko — APPROVED** [review]

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#build

**Review by ngmr — APPROVED** [review]

**Comment by jonhawkes** [comment]
!build

**Pull request merged**